src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -205,7 +205,23 @@ export const hasFilterKeyError = (
   return null;
 };
 
-const fieldDefined = (indexMappings: IndexMapping, key: string) => {
+export const fieldDefined = (indexMappings: IndexMapping, key: string) => {
   const mappingKey = 'properties.' + key.split('.').join('.properties.');
+  const potentialKey = get(indexMappings, mappingKey);
+
+  // If the `mappingKey` does not match a valid path, before returning null,
+  // we want to check and see if the intended path was for a multi-field
+  // such as `x.attributes.field.text` where `field` is mapped to both text
+  // and keyword
+  if (potentialKey == null) {
+    const indexOfLastProperties = mappingKey.lastIndexOf('properties');
+    const fieldMapping = mappingKey.substr(0, indexOfLastProperties);
+    // The 11 here refers to the length of `properties.`
+    const fieldType = mappingKey.substr(mappingKey.lastIndexOf('properties') + 11);
```

Comment:
my nit for readability, I will create a global `const propertiesAttribute = 'properties';` and then I will do that so I know exactly what's going on without reading the comment
```suggestion
    const fieldType = mappingKey.substr(mappingKey.lastIndexOf('properties') + `${propertiesAttribute}.`.length);
```